app/src/main/java/io/gnosis/safe/ui/settings/owner/ledger/LedgerDeviceListAdapter.kt

```diff
@@ -0,0 +1,58 @@
+package io.gnosis.safe.ui.settings.owner.ledger
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import androidx.viewbinding.ViewBinding
+import io.gnosis.safe.databinding.ItemLedgerDeviceBinding
+
+
+class LedgerDeviceListAdapter(private val listener: DeviceListener, private val forSigningOnly: Boolean = false) :
+    RecyclerView.Adapter<LedgerDeviceViewHolder>() {
+
+    private val items = mutableListOf<LedgerDeviceViewData>()
+
+    fun updateDeviceData(data: List<LedgerDeviceViewData>) {
+        items.clear()
+        items.addAll(data)
+        notifyDataSetChanged()
+    }
+
+    override fun onBindViewHolder(holder: LedgerDeviceViewHolder, position: Int) {
+        val device = items[position]
+        holder.bind(device, listener, position)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LedgerDeviceViewHolder {
+        return LedgerDeviceViewHolder(
+            ItemLedgerDeviceBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun getItemCount() = items.size
+
+    interface DeviceListener {
+        fun onDeviceClick(position: Int)
+    }
+}
+
+abstract class BaseOwnerViewHolder(
+    viewBinding: ViewBinding
+) : RecyclerView.ViewHolder(viewBinding.root)
+
+class LedgerDeviceViewHolder(private val viewBinding: ItemLedgerDeviceBinding) : BaseOwnerViewHolder(viewBinding) {
```

Comment:
Don't we need a DisabledLedgerDeviceViewHolder?

Reply:
What would this view holder show? What is a disabled ledger device?

Reply:
One that has already imported and named. It is possible to import more than one ledger owner address, isn't it?

Reply:
But this one is for the device list, and not the address list. You are most likely talking about next screen. Device can be either discovered - then it is shown, or it is not available - then it is missing in the list

Reply:
Sorry! I confused the two lists :-)